service/test/integration2_test.go

```diff
@@ -1132,7 +1140,9 @@ func TestSkipPrologue2(t *testing.T) {
 		callme3 := findLocationHelper(t, c, "main.callme3", false, 1, 0)[0]
 		callme3Z := uint64(clientEvalVariable(t, c, "main.callme3").Addr)
 		ver, _ := goversion.Parse(runtime.Version())
-		if ver.Major < 0 || ver.AfterOrEqual(goversion.GoVer18Beta) {
+
+		is64Bit := runtime.GOARCH == "amd64" || runtime.GOARCH == "arm64"
+		if (ver.Major < 0 || ver.AfterOrEqual(goversion.GoVer18Beta)) && is64Bit {
```

Comment:
I've compiled callme.go with GOARCH=386 and it does have a prologue, there's something wrong with prologue detection on i386 if this change is necessary.

Reply:
I find [unixGetG](https://github.com/go-delve/delve/pull/1884/commits/d7dc84ac9ea5289b7aee4632f9f983ab61b0dab0#diff-1aec94b2d00c8b6e87bc18b22ed320cfR32) should be {`mov`, `mov`}, I will remove [windowsGetG](https://github.com/go-delve/delve/pull/1884/commits/d7dc84ac9ea5289b7aee4632f9f983ab61b0dab0#diff-1aec94b2d00c8b6e87bc18b22ed320cfR33) temporarily.   

---  

I feel puzzled at  `something wrong with prologue detection`.  
```
19   func callme3() {
20 	 callme2()
21   }

// I disassme on 386
        .:0             0x80c566f       cc              int3
        callme.go:19    0x80c5670       658b0d00000000  mov ecx, dword ptr gs:[]
        callme.go:19    0x80c5677       8b89fcffffff    mov ecx, dword ptr [ecx+0xfffffffc]
        callme.go:19    0x80c567d       3b6108          cmp esp, dword ptr [ecx+0x8]
        callme.go:19    0x80c5680       7606            jbe 0x80c5688
        callme.go:20    0x80c5682       e819ffffff      call $main.callme2
        callme.go:21    0x80c5687       c3              ret
        callme.go:19    0x80c5688       e883c8fcff      call $runtime.morestack_noctxt
        callme.go:19    0x80c568d       ebe1            jmp $main.callme3
```
It is my understanding that `0x80c5670 ~ 0x80c5680`  for  line 19 are `stacksplit`'s codes. 
When  cmd `list main.callme3`, in fact it become line 20.  
So cmd `list callme.go:19` can only select `0x80c5670`.  

If I understand it wrong, please point it out. Thx.

Reply:
So I think this testcase may need special treatment for 386.

Reply:
Ok, you're right about this. However I think the test should be `runtime.GOARCH != "386"` rather than `runtime.GOARCH == "amd64" || runtime.GOARCH == "arm64"`.